catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/ProjectUploadRatingDialogTest.java

```diff
@@ -0,0 +1,172 @@
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.common.Constants.UPLOAD_RESULT_RECEIVER_RESULT_CODE;
+import static org.catrobat.catroid.ui.ProjectUploadActivity.NUMBER_OF_UPLOADED_PROJECTS;
+import static org.catrobat.catroid.ui.ProjectUploadActivity.PROJECT_DIR;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+public class ProjectUploadRatingDialogTest {
+
+	private ProjectUploadController projectUploadController;
+	private Bundle bundle;
+	private SharedPreferences sharedPreferences;
+	private static final String PROJECT_NAME = "projectName";
+
+	@Rule
+	public BaseActivityTestRule<ProjectUploadTestActivity> activityTestRule =
+			new BaseActivityTestRule<>(ProjectUploadTestActivity.class, false, false);
+
+	@Before
+	public void setUp() throws Exception {
+		Context context = InstrumentationRegistry.getTargetContext();
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		bundle = new Bundle();
+
+		Project project = new DefaultProjectCreator().createDefaultProject(PROJECT_NAME,
+				InstrumentationRegistry.getTargetContext(),	false);
+
+		Sprite firstSprite = new SingleSprite("firstSprite");
+		project.getDefaultScene().addSprite(firstSprite);
+		ProjectSaveTask.task(project, context);
+
+		Intent intent = new Intent();
+		intent.putExtra(PROJECT_DIR, project.getDirectory());
+		activityTestRule.launchActivity(intent);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testUploadControllerGetsCalled() {
+		uploadProject();
+
+		projectUploadController = activityTestRule.getActivity().getProjectUploadController();
+
+		verify(projectUploadController, times(1)).startUpload();
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void uploadSuccessRatingDialogShowing() {
+		uploadProject();
+
+		InstrumentationRegistry.getInstrumentation().runOnMainSync(() -> {
+			activityTestRule.getActivity().showUploadDialog();
+			activityTestRule.getActivity().onReceiveResult(UPLOAD_RESULT_RECEIVER_RESULT_CODE, bundle);
+		});
+
+		onView(withText(R.string.rating_dialog_rate_now))
+				.check(matches(isDisplayed()));
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void uploadFailRatingDialogNotShowing(){
+		uploadProject();
+
+		InstrumentationRegistry.getInstrumentation().runOnMainSync(() -> {
+			activityTestRule.getActivity().showUploadDialog();
+			activityTestRule.getActivity().onReceiveResult(0,bundle);
+		});
+
+		onView(withText(R.string.rating_dialog_rate_now))
+				.check(doesNotExist());
+	}
+
+	public void uploadProject(){
+		sharedPreferences.edit()
```

Comment:
since this happens in every test as the first thing, this probably should be in setUp().

at least the SharedPreferences call should be moved there.

and i rather have the code duplication of having the two espresso clicks in each test, than this. before looking up what the method really does i suspected anything, but two clicks on "next" in it when reading the tests above.